src/internal/predicates.mjs

```diff
@@ -29,3 +31,27 @@ export function isIterator(i){
 export function isArray(x){
   return Array.isArray(x);
 }
+
+export function hasMethod(method, x){
+  return Boolean(x) && typeof x[method] === 'function';
+}
+
+export function isFunctor(x){
+  return hasMethod(FL.map, x);
```

Comment:
What about `[]`, `{}`, `Math.abs`, and so on?

Reply:
Sadly, those don't have Fantasy Land implementations, and are exactly what I'm dropping support for with this commit (referring to it as "primitive dispatching"). Users who want to treat these values as Functors can use (and probably already are using) Ramda or Sanctuary. `map` and friends in Fluture can be thought of as operating on Futures and ConcurrentFutures (although they can work with any FL type).

Fluture currently supports it through sanctuary-type-classes, but at too great a cost. I am trying to shrink the Fluture core down to the essentials, and would like to [integrate with Sanctuary at a higher level](https://github.com/fluture-js/Fluture/projects/3#card-11455088).